### PR TITLE
[Hibernate] Add hasNext method to Pager api

### PR DESCRIPTION
**Pull request by acostama**
This allows a pager to be used similar to an iterator, with `hasNext` and `nextPage`

### CODE REVIEW

File: misk-hibernate/src/main/kotlin/misk/hibernate/pagination/Pager.kt [hunk 1]
```diff
@@ -24,12 +27,10 @@ fun <T : DbEntity<T>, R> Pager<T>.listAll(
   transform: (T) -> R
 ): List<R> {
   val results = mutableListOf<R>()
-  var offset: Offset?
```
Comment (acostama) [inline misk-hibernate/src/main/kotlin/misk/hibernate/pagination/Pager.kt:27]: this way, it isn't necessary to track the offset

File: misk-hibernate/src/main/kotlin/misk/hibernate/pagination/RealPager.kt [hunk 2]
```diff
@@ -39,4 +41,6 @@ internal class RealPager<T : DbEntity<T>, Q: Query<T>>(
       contents to false
     }
   }
+
+  override fun hasNext() = hasMore
```
Comment (zcross) [inline misk-hibernate/src/main/kotlin/misk/hibernate/pagination/RealPager.kt:45]: Can we pick one phrasing and use it uniformly?

File: misk-hibernate/src/test/kotlin/misk/hibernate/pagination/RealPagerTest.kt [hunk 3]
```diff
@@ -84,6 +84,58 @@ class RealPagerTest {
     assertThat(actualCharacterNames).containsExactlyElementsOf(expectedCharacterNames)
   }
 
+  @Test fun `hasNext always true on first page`() {
```
Comment (zcross) [inline misk-hibernate/src/test/kotlin/misk/hibernate/pagination/RealPagerTest.kt:87]: Thanks for the comprehensive tests!

File: misk-hibernate/src/main/kotlin/misk/hibernate/pagination/RealPager.kt (hunk 2 above)
Comment (acostama) [reply misk-hibernate/src/main/kotlin/misk/hibernate/pagination/RealPager.kt]: changed

**Pull request merged**